app/components/NavButtons/index.js

```diff
@@ -5,13 +5,25 @@ import FontAwesome from 'react-fontawesome';
 import styles from './styles.scss';
 
 const NavButtons = props => {
+
+  let enableBackButton = true;
+  let enableForwardButton = true;
+
+  if (props.historyCurrentIndex == 1) {
+    enableBackButton = false;
+  }
+
+  if (props.historyCurrentIndex == (props.historyLength - 1)) {
```

Comment:
This one and the condition above could be encapsulated in their own functions for simplicity.